packages/rspack/src/compilation.ts

```diff
@@ -615,6 +616,71 @@ export class Compilation {
 		});
 	}
 
+	get chunks() {
+		var stats = this.getStats().toJson({
+			all: false,
+			chunks: true,
+			chunkModules: true,
+			reasons: true
+		});
+		const chunks = stats.chunks?.map(chunk => {
+			return {
+				...chunk,
+				name: chunk.names.length > 0 ? chunk.names[0] : "",
+				modules: this.__internal__getAssociatedModules(chunk)
+			};
+		});
+		return chunks;
+	}
+
+	/**
+	 * Get the associated `modules` of an given chunk.
+	 *
+	 * Note: This is not a webpack public API, maybe removed in future.
+	 *
+	 * @internal
+	 */
+	__internal__getAssociatedModules(chunk: JsStatsChunk): any[] | undefined {
+		let modules = this.getModules();
+		let moduleMap: Map<string, JsModule> = new Map();
+		for (let module of modules) {
+			moduleMap.set(module.moduleIdentifier, module);
+		}
+		return chunk.modules?.flatMap(chunkModule => {
+			let jsModule = this.__internal__findJsModule(
+				chunkModule.issuer ?? chunkModule.identifier,
+				moduleMap
+			);
+			return {
+				...jsModule
+				// dependencies: chunkModule.reasons?.flatMap(jsReason => {
```

Comment:
I have tested the `webpack-licence-plugin` could work fine without `dependencies` which could reduce a lot of calculation.

Reply:
Yes, it's creating the license file properly. But without these dependencies, the final license file has fewer licenses than the final generated by the Webpack bundler.

The licenses file includes all first-level included dependencies but not for all second-level and the following.

Reply:
But anyway, the current implementation is a great achievement.

Reply:
This is just a trade-off, in a big project which has 10000+ modules, we might have performance issues if we calculate all the dependencies, let me discuss with rspack team.